docs/reusing-typecheck-results.md

```diff
@@ -0,0 +1,317 @@
+---
+title: Reusing compilation results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing compilation results between compiler and tooling runs
+
+Caching and reusing compilation results between compiler and tooling runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+## Motivation
+
+- **Performance**. Things like recomputing type information for large projects are time-consuming. By caching the results, subsequent compilation can skip some steps, leading to faster builds.
+
+- **Better IDE cold start**. IDEs can provide faster IntelliSense, code navigation, and other features if they can quickly access cached compilation information.
+
+- **Error reduction**. Consistent compilation results help avoid scenarios where slight differences between runs could cause different errors or warnings to be produced.
+
+### Example real-world scenarios
+
+The optimization will benefit the most in large and not very interdependent projects. For instance, a 100-files test project where a single test is changed. MSBuild will mark the project for recompilation but thanks to the improvement only one file would be recompiled in the full-blown manner.
+
+The technique should also help when MSBuild generally incorrectly marks projects for rebuild. The probably doesn't happen often though.
```

Comment:
It's done on timestamps, which this feature will likely be doing the same.
=> better to delete this